kratos/containers/master_slave_relation.h

```diff
@@ -0,0 +1,201 @@
+    //double ConstantUpdate() const { return mConstantUpdate; }
+    IndexType SlaveEquationId() const { return this->Id(); }
+
+
+
+    // Get number of masters for this slave
+    std::size_t GetNumberOfMasters() const
+    {
+        return mMasterDataSet.size();
+    }
+
+
+    /**
+     * this determines the master equation IDs connected to this constraint
+     * @param rResult the elemental equation ID vector
+     * @param rCurrentProcessInfo the current process info instance
+     */
+    virtual void EquationIdVector(EquationIdVectorType& rSlaveEquationIds,
+                                  EquationIdVectorType& rMasterEquationIds,
+                                  ProcessInfo& rCurrentProcessInfo)
+    {
+        if (rSlaveEquationIds.size() != 0)
+            rSlaveEquationIds.resize(1);
+
+        if (rMasterEquationIds.size() != 0)
+            rMasterEquationIds.resize(this->GetNumberOfMasters());
+
+        rSlaveEquationIds[0] = this->SlaveEquationId();
+
+        auto master_it = mMasterDataSet.begin();
+        for (IndexType i=0; i<this->GetNumberOfMasters(); i++)
+            rMasterEquationIds[i] = (*(++master_it)).first;
+    }
+
+    /**
+     * this is called during the assembling process in order
+     * to calculate all elemental contributions to the global system
+     * matrix and the right hand side
+     * @param rTransformationMatrix the elemental left hand side matrix
+     * @param rConstant the elemental right hand side
+     * @param rCurrentProcessInfo the current process info instance
+     */
+    virtual void CalculateLocalSystem(MatrixType& rTransformationMatrix,
+                                      VectorType& rConstantVector,
+                                      ProcessInfo& rCurrentProcessInfo)
+    {
+      if (rTransformationMatrix.size1() != 0)
+      {
+    	rTransformationMatrix.resize(1, this->GetNumberOfMasters(), false);
+      }
+
+      if (rConstantVector.size() != 0)
+      {
+    	rConstantVector.resize(1, false);
+      }
+
+        auto master_it = mMasterDataSet.begin();
+        for (IndexType i=0; i<this->GetNumberOfMasters(); i++)
+            rTransformationMatrix(0,i) = (*(++master_it)).second;
+
+      rConstantVector(0) = Constant();
+    }
+
+    void PrintInfo(std::ostream& rOutput) const override
+    {
+        rOutput << "##############################" << std::endl;
+        rOutput << "SlaveEquationId :: " << SlaveEquationId() << std::endl;
+        rOutput << "Constant :: " << Constant() << std::endl;
+        int index = 0;
+        rOutput << "############################## :: Masters" << std::endl;
+        for (auto &master : mMasterDataSet)
+        {
+            rOutput << index << " Master  equation id :: " << master.first << ", weight :: " << master.second << std::endl;
+            index++;
+        }
+        rOutput << "##############################" << std::endl;
+    }
+
+    void Clear()
+    {
+        mMasterDataSet.clear();
```

Comment:
why not also the data?

Reply:
There is no data in this class except this ´´´mMasterDataSet´´´.